src/index.js

```diff
@@ -13,7 +13,25 @@ import AppContextProvider from './AppContext';
 
 import Theme from './Theme';
 
-const client = new ApolloClient({ uri: 'https://graphql.hurumap.org/graphql' });
+const client = new ApolloClient({
+  uri: 'https://graphql.hurumap.org/graphql',
+  onError: ({ graphQLErrors, networkError }) => {
+    if (process.env.NODE_ENV === 'development') {
+      if (graphQLErrors) {
+        graphQLErrors.map(({ message, locations, path }) =>
+          // eslint-disable-next-line no-console
+          console.error(
+            `[GraphQL error]: Message: ${message}, Location: ${locations}, Path: ${path}`
+          )
```

Comment:
Lets send the error message as `state` i.e. `history.replace(path, [state])` or `history.push(path, [state])` to the error page and the error page can decide to either just `console.log` it or show a `div` with the error message (in `dev` env)... But all logging etc. should happen from there.